### PR TITLE
Bugfix/docs: correct and expand smoke test documentation

### PR DESCRIPTION
**Pull request by tldahlgren**
The smoke test documentation referenced a directory property used in a few preliminary packages rather than the actual directory property.  This PR corrects the property and expands on the smoke test documentation to include describing more options and use of the package `test` subdirectory for custom files.

### CODE REVIEW

**Comment by adamjstewart** [comment]
Can you integrate #20054 into this? For example, adding `.. _cmd-spack-test-list:` before the corresponding docs section will add a link to that section under https://spack.readthedocs.io/en/latest/command_index.html#spack-test-list.

File: lib/spack/docs/packaging_guide.rst [hunk 1]
```diff
@@ -3956,111 +3956,192 @@ Install Tests
...
+Test suites can be named using the ``--alias`` option. Unaliased
+Test suites will use the content hash of their specs as their name.
 
-.. code-block:: yaml
+Some of the more commonly used debugging options are:
 
-   config:
-     test_stage: /path/to/stage
+- ``--fail-fast`` stops testing each package after the first failure
+- ``--fail-first`` stops testing packages after the first failure
```
Comment (alalazo) [inline lib/spack/docs/packaging_guide.rst:4111]: Just a note that "failed first" is used in other testing frameworks, like `pytest`, to run all tests re-ordering the last failures to run first.

**Review by alalazo — APPROVED** [review]
LGTM 👍

**Comment by tldahlgren** [comment]
> LGTM 👍

Thanks!

**Pull request merged**